### PR TITLE
DOC: Fix dev docs in release branch (v6.0.x)

### LINKED ISSUE #15594: [v6.0.0rc1] docs failure because of dev docs
### Description

The build on the 6.0.0rc1 tag failed :
https://readthedocs.org/api/v2/build/22507043.txt
due to
```
/home/docs/checkouts/readthedocs.org/user_builds/astropy/checkouts/v6.0.0rc1/docs/index.rst:34: WARNING: toctree contains reference to excluded document 'development/index'
...
/home/docs/checkouts/readthedocs.org/user_builds/astropy/checkouts/v6.0.0rc1/docs/index.rst:93: WARNING: 'any' reference target not found: development/index
```
probably related to the reorganization of toctree/index pages with the new theme, and the magic happening to exclude dev docs from releases.

### Expected behavior

_No response_

### How to Reproduce

1. Get package from '...'
2. Then run '...'
3. An error occurs.

```python
# Put your Python code snippet here.
```

### Versions

import platform; print(platform.platform())
import sys; print("Python", sys.version)
import astropy; print("astropy", astropy.__version__)
import numpy; print("Numpy", numpy.__version__)
import erfa; print("pyerfa", erfa.__version__)
import scipy; print("Scipy", scipy.__version__)
import matplotlib; print("Matplotlib", matplotlib.__version__)

### PR DESCRIPTION
**Pull request by pllim**
### Description

This pull request is to fix dev docs in release branch so it works for non-dev branch with pydata-sphinx-theme. 

I think this will do it but I am not sure how to test it without doing another RC tag. The trigger only differentiates between dev and non-dev tags.

If merged, this should be *forward-ported* back to `main`.

Fixes #15594

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

**Comment by pllim** [comment]
pre-commit.ci autofix

File: docs/conf.py [hunk 1]
```diff
@@ -226,10 +226,20 @@
...
+        "development/a*",
+        "development/b*",
+        "development/c*",
+        "development/d*",
+        "development/r*",
+        "development/s*",
+        "development/t*",
+        "development/v*",
+        "development/w*",
+    ]
```
Comment (saimn) [inline docs/conf.py:242]: Seems a bit fragile... Maybe instead we can move `development/index` to `index_dev.rst` ?

Comment (pllim) [reply docs/conf.py:242]: I can try... I remember pydata-sphinx-theme left/right menu being highly influenced by the layout but I don't remember this moving the index to this folder was part of that or not.

**Comment by pllim** [comment]
Closing in favor of https://github.com/astropy/astropy/pull/15596

**Pull request closed without merging**